sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerClient.java

```diff
@@ -175,6 +183,55 @@ public PartitionProperties getPartitionProperties(String partitionId) {
         return new IterableStream<>(map);
     }
 
+    /**
+     * Receives a batch of {@link PartitionEvent events} from all partitions. The batch is returned after
+     * {@code maximumMessageCount} events are received or after the
+     * {@link RetryOptions#getTryTimeout() operation timeout} has elapsed.
+     *
+     * @param maximumMessageCount The maximum number of messages to receive in this batch.
+     *
+     * @return A set of {@link PartitionEvent} that was received. The iterable contains up to
+     *     {@code maximumMessageCount} events.
+     *
+     * @throws IllegalArgumentException if {@code maximumMessageCount} is less than 1.
+     */
+    public IterableStream<PartitionEvent> receive(int maximumMessageCount) {
+        return receive(maximumMessageCount, timeout);
+    }
+
+    /**
+     * Receives a batch of {@link PartitionEvent events} from all partitions. The batch is returned after
+     * {@code maximumMessageCount} events are received or after {@code maximumWaitTime} has elapsed.
+     *
+     * @param maximumMessageCount The maximum number of messages to receive in this batch.
+     * @param maximumWaitTime The maximum amount of time to wait to build up the requested message count for the
+     *     batch; if not specified, the default wait time specified when the consumer was created will be used.
+     *
+     * @return A set of {@link PartitionEvent} that was received. The iterable contains up to
+     *     {@code maximumMessageCount} events.
+     *
+     * @throws NullPointerException if {@code maximumWaitTime} is null.
+     * @throws IllegalArgumentException if {@code maximumMessageCount} is less than 1 or {@code maximumWaitTime} is
+     *     zero or a negative duration.
+     */
+    public IterableStream<PartitionEvent> receive(int maximumMessageCount, Duration maximumWaitTime) {
+        Objects.requireNonNull(maximumWaitTime, "'maximumWaitTime' cannot be null.");
+
+        if (maximumMessageCount < 1) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("'maximumMessageCount' cannot be less than 1."));
+        } else if (maximumWaitTime.isNegative() || maximumWaitTime.isZero()) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("'maximumWaitTime' cannot be zero or less."));
+        }
+
+        final Flux<PartitionEvent> events = Flux.create(emitter -> {
+            queueWork(RECEIVE_ALL_KEY, maximumMessageCount, maximumWaitTime, emitter);
+        });
+
+        return new IterableStream<>(events);
```

Comment:
The `map` operation can be removed in `receive(String partitionId, int maximumMessageCount, Duration maximumWaitTime)` overload as well.